src/backends/github/API.js

```diff
@@ -20,70 +21,47 @@ export default class API {
   }
 
   isCollaborator(user) {
-    return this.request('/user/repos').then((repos) => {
-      let contributor = false
-      for (const repo of repos) {
-        if (repo.full_name === this.repo && repo.permissions.push) contributor = true;
-      }
-      return contributor;
-    }).catch((error) => {
-      console.error("Problem with response of /user/repos from GitHub");
-      throw error;
-    })
+    return this.request('/user/repos').then(repos =>
+      Object.keys(repos).some(
+        key => (repos[key].full_name === this.repo) && repos[key].permissions.push
+      )
+    );
   }
 
   requestHeaders(headers = {}) {
-    const baseHeader = {
-      "Content-Type": "application/json",
+    return {
       ...headers,
+      ...(this.token ? { Authorization: `token ${ this.token }` } : {}),
+      "Content-Type": "application/json",
     };
-
-    if (this.token) {
-      baseHeader.Authorization = `token ${ this.token }`;
-      return baseHeader;
-    }
-
-    return baseHeader;
-  }
-
-  parseJsonResponse(response) {
-    return response.json().then((json) => {
-      if (!response.ok) {
-        return Promise.reject(json);
-      }
-
-      return json;
-    });
   }
 
   urlFor(path, options) {
-    const cacheBuster = new Date().getTime();
-    const params = [`ts=${cacheBuster}`];
-    if (options.params) {
-      for (const key in options.params) {
-        params.push(`${ key }=${ encodeURIComponent(options.params[key]) }`);
-      }
-    }
-    if (params.length) {
-      path += `?${ params.join("&") }`;
-    }
-    return this.api_root + path;
+    const cacheBuster = `ts=${ new Date().getTime() }`;
+    const encodedParams = options.params
+          ? Object.entries(options.params).map(
+            ([key, val]) => `${ key }=${ encodeURIComponent(val) }`)
+          : [];
+    return `${ this.api_root }${ path }?${ [cacheBuster, ...encodedParams].join("&") }`;
```

Comment:
Can we make use of modern API here (to avoid re-inventing the wheel)?
Specifically:

- [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) for the query string
- [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL) for the merging of parts (root, path and query string)

Reply:
Just a note: `URLSearchParams` isn't supported in MS Edge <17. Otherwise, I think this is a great idea.